service/db.hpp

```diff
@@ -80,370 +93,309 @@ namespace raven
         }
         catch (const sqlite::sqlite_exception &error) {
             DLOG_F(ERROR, "error: %s, from sql: %s", error.what(), error.get_sql().c_str());
-            answer.request_state = convert_request_state.at(request_state::db_error);
+            state = db_state::sql_error;
         }
         catch (const std::exception &error) {
             DLOG_F(ERROR, "%s", error.what());
-            answer.request_state = convert_request_state.at(request_state::db_error);
+            state = db_state::fatal_error;
         }
-        return answer;
+        return data;
     }
 
-    config_include_answer_db config_include(const config_include& config_include_data) noexcept
+    void update_config(const json::json &updated_data, config_id_st id) noexcept
     {
-        //TODO: Communication.md has changed, need to be modified
+        /*
+         * Update the config corresponding to the id with the json data passed as parameter
+         *
+         * In case an error occur, the state will be set accordingly (`unknow_config_id`, `sql_error`, `fatal_error`)
+         *
+         */
+
         LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
-        config_include_answer_db answer{request_state::success, 0u};
         try {
-            auto functor_receive_data = [this, &config_include_data, &answer](const std::string json_text) {
-                auto json_data = json::json::parse(json_text);
-                DLOG_F(INFO, "json before update: %s", json_data.dump().c_str());
-                auto& array_other_config = json_data["OTHER_CONFIG"];
-                array_other_config.push_back(config_include_data.src_id.value());
-                std::sort(begin(array_other_config), end(array_other_config));
-                auto last = std::unique(begin(array_other_config), end(array_other_config));
-                array_other_config.erase(last, end(array_other_config));
-                DLOG_F(INFO, "json after update: %s", json_data.dump().c_str());
-                answer.nb_configs = array_other_config.size();
-                this->execute_statement(update_config_text_from_id_statement,
-                                        json_data.dump().c_str(),
-                                        config_include_data.id.value());
-            };
-            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement,
-                config_include_data.id.value());
-            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement,
-                                                                 config_include_data.src_id.value());
-            execute_statement(select_config_from_id_statement, config_include_data.id.value()) >> functor_receive_data;
+            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement, id.value());
+            execute_statement(update_config_text_from_id_statement, updated_data.dump().c_str(), id.value());
```

Comment:
`.dump().c_str()` is suspicious, don't remember what the conclusion was last time.

Reply:
don't remember what we said about that last time but you're right it is suspicious